bundler/lib/bundler/source/git/git_proxy.rb

```diff
@@ -53,10 +60,16 @@ class GitProxy
         def initialize(path, uri, options = {}, revision = nil, git = nil)
           @path     = path
           @uri      = uri
-          @branch   = options["branch"]
           @tag      = options["tag"]
+          @branch   = options["branch"]
           @ref      = options["ref"]
-          @explicit_ref = branch || tag || ref
+          @explicit_ref = if @tag
+            raise AmbiguousGitReference.new(options) if @branch || @ref
+
+            @tag
+          else
+            @ref || @branch
+          end
```

Comment:
Style tweak. This branch is complicated. How about something more like: 

```suggestion
          raise AmbiguousGitReference.new(options) if @tag && (@branch || @ref)
          @explicit_ref = @tag || @ref || @branch
```

Reply:
ok, I'm fine with that, but can you move the assignment into the `if`? I'll update my suggestion above to clarify what I mean. Something about the nested raise/if within an nested assignment if. The logical purity is there, but the assignment purity is obscured.

Reply:
Another way of explaining my discomfort with the "simple" way is that these two lines "look" like a bug to someone unfamiliar with the reasoning behind the logic.
```
          raise AmbiguousGitReference.new(options) if @tag && (@branch || @ref)
          @explicit_ref = @tag || @ref || @branch
```
If I saw this in business logic app code I'd ask for a comment to explain, or a refactor.  The logic as-is needs no explanation because the branching logic isn't collapsed.

Reply:
You make a good point. I felt a twinge of the same feeling when I collapsed it to the above. My real beef is with the assignment outside a complicated if. I'd rather duplication than obscuration.

Reply:
Though I do know that with some RuboCop configs that'll be flagged, we'll see about this one.

Reply:
Amended commit. 😄

Reply:
RuboCop did not flag on either of the rules I thought it might. 👍

Reply:
Rubucop has shown us compassion today.